src/commands/DensityProfile.cpp

```diff
@@ -166,22 +299,61 @@ void DensityProfile::accumulate(const chemfiles::Frame& frame, Histogram<double>
 
 void DensityProfile::finish(const Histogram<double>& profile) {
     std::ofstream outfile(options_.outfile, std::ios::out);
-    auto axis = axis_.get_coordinates();
-    if(outfile.is_open()) {
+    auto axis_x = axis_x_.get_coordinates();
+    auto axis_y = axis_y_.get_coordinates();
+    if (outfile.is_open()) {
         outfile << "# Density profile in trajectory " << AveCommand::options().trajectory << std::endl;
-        outfile << "# along axis " << axis[0] << ' ' << axis[1] << ' ' << axis[2] << std::endl;
+        outfile << "# along axis " << axis_x[0] << ' ' << axis_x[1] << ' ' << axis_x[2] << std::endl;
+        if (n_axis_ == 2) {
+            outfile << "# and along axis " << axis_y[0] << ' ' << axis_y[1] << ' ' << axis_y[2] << std::endl;
+        }
         outfile << "# Selection: " << options_.selection << std::endl;
 
-        double dr = profile.bin_size();
-        for (size_t i=0; i<profile.size(); i++){
-            if (options_.type_profile == "along_axis") {
-                outfile << profile.min() + i * dr << "  " << profile[i] << "\n";
-            } else if (options_.type_profile == "radial") {
-                auto r = profile.min() + (i + 0.5) * dr;
-                if (r == 0) {
-                    r = dr / 1000; // use a small r compared to dr to avoid Nan in the output
+        double dx = profile.dx();
+        double dy = profile.dy();
+        if (n_axis_ == 1) {
+            double min_x = profile.min_x();
+            for (size_t i=0; i<profile.size(); i++){
+                if (options_.type_profile[0] == 1) {
+                    outfile << min_x + i * dx << "  " << profile[i] << "\n";
+                } else if (options_.type_profile[0] == 2) {
+                    auto r = min_x + (i + 0.5) * dx;
+                    if (r == 0) {
```

Comment:
Because you are doing `r = dr * (i + 0.5)`, this should never be true, isn't it?